pkg/server/skydb/access.go

```diff
@@ -126,3 +130,192 @@ func (acl RecordACL) Accessible(userinfo *UserInfo, level RecordACLLevel) bool {
 
 	return accessible
 }
+
+// FieldAccessMode is the intended access operation to be granted access
+type FieldAccessMode int
+
+const (
+	// ReadFieldAccessMode means the access mode is for reading
+	ReadFieldAccessMode FieldAccessMode = iota + 1
+
+	// WriteFieldAccessMode means the access mode is for writing
+	WriteFieldAccessMode
+
+	// DiscoverFieldAccessMode means the access mode is for discovery
+	DiscoverFieldAccessMode
+
+	// CompareFieldAccessMode means the access mode is for query
+	CompareFieldAccessMode
+)
+
+// FieldACL contains all field ACL rules for all record types
+type FieldACL struct {
+	wildcardRecordType FieldACLEntryList
+	recordTypes        map[string]FieldACLEntryList
+}
```

Comment:
It is necessary to keep 1 dictionary + 1 array ? Or we can improve the sorting methods of `FieldACLEntryList` to keep it just 1 array ?